core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -3394,6 +3399,10 @@ protected function reviseTable()
 			{
 				$objIds = $this->Database->execute("SELECT c.id FROM " . $this->strTable . " c LEFT JOIN " . $ptable . " p ON c.pid=p.id WHERE c.ptable='" . $ptable . "' AND p.id IS NULL");
 			}
+			elseif ($ptable == $this->strTable)
+			{
+				$objIds = $this->Database->execute('SELECT c.id FROM ' . $this->strTable . ' c LEFT JOIN ' . $ptable . ' p ON c.pid=p.id WHERE p.id IS NULL AND c.pid > 0');
```

Comment:
I do not understand the `c.pid > 0` part. How does it solve the issue?

Reply:
The root pages have `c.pid=0` and therefore no parent node, but `reviseTable()` should not delete the root pages but only pages with a set `pid` that points to a non-existing parent.